habitat/core/env.py

```diff
@@ -423,3 +424,232 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close()
+
+
+class MultiTaskEnv(Env):
+    r"""Environment base class extension designed to handle a stream-line of tasks.
+
+    The major :ref:`task` component is hereby intended as current task with which
+    the agent is currently interacting.
+    All tasks handled by the MultiTaskEnv are available through the :ref:`tasks` object
+    property.
+
+    The environment abstracts away the serving of multiple tasks under
+    the scheduling preferences expressed in the configuration.
+    """
+    _tasks: List[EmbodiedTask]
+    DEFAULT_EPS_CHANGE = 10
+
+    def __init__(
+        self, config: Config, dataset: Optional[Dataset] = None
+    ) -> None:
+        """Constructor. Mimics the :ref:`Env` constructor.
+
+        Args:
+            :param config: config for the environment. Should contain id for
+            simulator and ``task_name`` for each task, which are passed into ``make_sim`` and
+            ``make_task``.
+            :param dataset: reference to dataset used for first task instance level
+            information. Can be defined as :py:`None` in which case
+            dataset will be built using ``make_dataset`` and ``config``.
+        """
+        # we let superclass instantiate first task as to not break/change habitat code
+        super().__init__(config, dataset=dataset)
+        # instatiate other tasks
+        self._tasks = [self._task]
+        self._curr_task_idx = 0
+        for task in config.TASKS[1:]:
```

Comment:
Wait, so is self.TASKS[0] ignored then and just replace by self._task? If so, why bother including it all? Seems error prone.

Reply:
Oh I see, it mutated the TASK_CONFIG in the config loader. Still prone to error if a config.TASKS also has a config.TASK defined and people mutate it (as it would if you are printing out the config to rerun experiments). We probably want some error or at least a warning if the config value.TASK is being overwritten. It shouldn't be a hard error though if it's mutated like this. 

A lot of code directly accesses the underlying task config though so I definitely see the potential issue with this. Your way might be the best solution for now, will consult with other contributors.

Reply:
Exactly, in order to subclass `Env` and make use of the default task init logic without 'wasting' the super() call I had to overwrite config.TASK with the first config.TASKS entry, so that `self._task` remains as the current task variable.
Not sure whether we can have TASK and TASKS co-exist in a meaningful way in the config file, so I'll add a warning message as you suggested when overwriting the existing config.TASK.

Reply:
I think it would be better suited to have the constructor here do the logic of moving overwriting `config.TASK` with `config.TASKS[0]`.  If folks end up programmatically changing the config, I think this will be less confusing.